### ISSUE TITLE
http-subscription-server not routing http

### ISSUE THREAD

**Issue by ioxie** [unbit/uwsgi#1402]
Hello,

I’m trying to do some kind of load balancing with http-subscription-server for some web applications that do not have support for uwsgi protocol and I need to route http. But this settings are not working:

`uwsgi --http :80 --http-subscription-server=10.0.0.1:4040 --master`

Trying with a simple app:

`uwsgi --http 10.0.0.8:8080 --subscribe2 server=10.0.0.1:4040,key=test.local,addr=10.0.0.8:8080  -M --manage-script-name --mount /=app:app`

The app is subscribed successfully to the subscription-server, but requests are not routed.

Testing from host 10.0.0.1:
Curl to http://test.local/, it has a timeout with empty response (127.0.0.1 test.local is present hosts file). And direct curl to http://10.0.0.8:8080/ responded ok.

Similar results when running the app with http-socket:
`uwsgi --http-socket 10.0.0.8:8080 --subscribe2 server=…`

If running the app with --socket (uwsgi --socket 10.0.0.8:8080…), curl to http://test.local/ works but of course it doesn’t work directly anymore and this is not desired.

I’m using uWSGI 2.0.14 installed with pip.

Am I doing something wrong?

**Comment by unbit** [comment]
Hi, http to http is only supported in 2.1 branch (add proto=http to subscribe2)

**Comment by unbit** [comment]
Eventually there is the "dumb" mode where the connection is downgraded to a basic tcp one, to force a subscription in dumb mode just use the magic modifier1 "123":

```
--subscribe2 server=10.0.0.1:4040,key=test.local,addr=10.0.0.8:8080,modifier1=123
```

**Comment by ioxie** [comment]
I got the master and used the proto=http option. Thank You.

Another hiccup that I have is that the subscriber app needs to be aware of the listening port of the subscription server and to register with the port in the key (ex. test.local:8080) if the subscription server listens of anything else than 80. That means the router check's the whole request <hostname>:<port> against the subscribed key.
I managed to work with it and everything is great, I just wanted to say that it might be nice to split port checking from the hostname.

Thank you again for the suggestions.
